File: Leetcode/sbgo/Leetcode334-递增的三元子序列.go
package sbgo

func increasingTriplet(nums []int) bool {
	const INF = int(^uint(0) >> 1)
	f := []int{INF, INF}
	for _, num := range nums {
		k := 2
		for k > 0 && f[k-1] >= num {
			k--
		}
		if k == 2 {
			return true
		}
		f[k] = num
	}
	return false
}

File: Leetcode/JS/Leetcode334-递增的三元子序列.js
var increasingTriplet = function(nums) {
    const INF = Number.MAX_VALUE;
    // console.log(INF);
    const f = [INF,INF];
    for(const num of nums){
        let k = 2;
        while(k > 0 && f[k - 1] >= num) k--;
        if(k == 2) return true;
        f[k] = num;
    }
    return false;
};